File: maximum_nesting_depth_of_the_parentheses.py
def maxDepth(self, s: str) -> int:
        count = 0
        maxCount = 0
        for i in range(len(s)):
            if s[i] == "(":
                count += 1
                maxCount = max(maxCount, count)
            elif s[i] == ")":
                count -= 1
        return maxCount

print(maxDepth("(1+(2*3)+((8)/4))+1")) # Output: 3

File: maximum_nesting_depth_of_the_parentheses.js
var maxDepth = function (s) {
  let count = 0,
    maxCount = 0;
  for (let i = 0; i < s.length; i++) {
    if (s[i] === "(") count++;
    if (s[i] === ")") count--;
    maxCount = Math.max(maxCount, count);
  }
  return maxCount;
};

console.log(maxDepth("(1+(2*3)+((8)/4))+1")); //Output: 3
